arduino-ide-extension/src/node/config-service-impl.ts

```diff
@@ -182,7 +182,11 @@ export class ConfigServiceImpl
       });
       const model = (yaml.safeLoad(content) || {}) as DefaultCliConfig;
       this.logger.info(`Loaded CLI configuration: ${JSON.stringify(model)}`);
-      if (model.directories.data && model.directories.user) {
+      if (
+        model.directories &&
```

Comment:
If `directories` is optional, I propose reflecting it on the type:
https://github.com/arduino/arduino-ide/blob/2e78e96b75b8c8c3971f6b115eec558ba84d0317/arduino-ide-extension/src/node/cli-config.ts#L98

Reply:
I think `directories` can be undefined only on the initial parse of the current config. 
We can manipolate the loaded config  object to assure the directories are alway defined  object returned by `loadCliConfig` is always of type `DefaultCliConfig`.

That's the current logic: 

1. Load current cli config as type `CliConfig` (optional `directories`).
2. Check `directories` value:
    1. If defined, return config as `DefaultCliConfig` type
    2. If undefined, call `getFallbackCliConfig` to get default directories, merge them to current config and return the object as `DefaultCliConfig`